sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -201,6 +202,7 @@ Connection getConnection() throws Exception {
   public abstract static class Read<T> extends PTransform<PBegin, PCollection<T>> {
     @Nullable abstract DataSourceConfiguration getDataSourceConfiguration();
     @Nullable abstract String getQuery();
+    @Nullable abstract List<Object> getParameters();
```

Comment:
Maybe a `SortedMap` containing parameter `java.sql.Type` and value would be better (in the order used in the query). A `List<Object>` is error prone IMHO.